python/pyspark/sql/tests/test_catalog.py

```diff
@@ -16,20 +16,21 @@
 #
```

Comment:
The changes made in reused PySpark tests are two:

1. Do not check the specific type of exception. We should fix it to check properly (SPARK-41715)
2. Explicit `collect` on `sql` because `sql` in Spark Connect is lazy (which actually I think is an issue).